server/utils/checkYourAnswersUtils.ts

```diff
@@ -0,0 +1,207 @@
+import { Cas2Application as Application } from '@approved-premises/api'
+import { SummaryListItem, FormSection } from '@approved-premises/ui'
+import Apply from '../form-pages/apply/index'
+import CheckYourAnswers from '../form-pages/apply/check-your-answers'
+import paths from '../paths/apply'
+import { getQuestions } from '../form-pages/utils/questions'
+import { nameOrPlaceholderCopy } from './utils'
+import { formatLines } from './viewUtils'
+import { DateFormats } from './dateUtils'
+
+export const checkYourAnswersSections = (application: Application) => {
+  const sectionsWithAnswers = getSectionsWithAnswers()
+
+  return sectionsWithAnswers.map(section => {
+    return {
+      title: section.title,
+      tasks: section.tasks.map(task => {
+        return {
+          id: task.id,
+          title: task.title,
+          rows: getTaskAnswersAsSummaryListItems(task.id, application),
+        }
+      }),
+    }
+  })
+}
+
+export const getTaskAnswersAsSummaryListItems = (task: string, application: Application): Array<SummaryListItem> => {
+  const items: Array<SummaryListItem> = []
+
+  const questions = getQuestions(nameOrPlaceholderCopy(application.person))
+
+  const pagesKeys = getPages(application, task)
+
+  pagesKeys.forEach(pageKey => {
+    addPageAnswersToItemsArray({
+      items,
+      application,
+      task,
+      pageKey,
+      questions,
+    })
+  })
+
+  return items
+}
+
+export const addPageAnswersToItemsArray = (params: {
+  items: Array<SummaryListItem>
+  application: Application
+  task: string
+  pageKey: string
+  questions: Record<string, unknown>
+}) => {
+  const { items, application, task, pageKey, questions } = params
+  const questionKeys = Object.keys(application.data[task][pageKey])
+  if (containsQuestions(questionKeys)) {
+    questionKeys.forEach(questionKey => {
+      const item = summaryListItemForQuestion(application, questions, task, pageKey, questionKey)
```

Comment:
Not a pressing immediate concern I suppose, but do you see opportunities in this work to introduce simplified function signatures. e.g. to have an object or data structure which represents a question so you could call `summaryListItemForQuestion() with fewer params`:

```ts
summaryListItemForQuestion(question)
```
and 

```ts
getAnswer(question)
```